libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -226,17 +245,33 @@ std::string YulUtilFunctions::revertWithError(
 {
 	solAssert((!_posVar.empty() && !_endVar.empty()) || (_posVar.empty() && _endVar.empty()));
 	bool const needsNewVariable = !_posVar.empty() && !_endVar.empty();
+	bool needsAllocation = true;
+
+	if (std::optional<size_t> size = staticEncodingSize(_parameterTypes))
+		if (
+			ranges::all_of(_parameterTypes, [](auto const* type) {
+				solAssert(!dynamic_cast<InaccessibleDynamicType const*>(type));
+				return type && type->isValueType();
+			})
+		)
+			needsAllocation = *size + 4 > CompilerUtils::generalPurposeMemoryStart;
```

Comment:
out of curiosity, why plus 4?

Reply:
Size of error selector.

Reply:
Got it. So maybe it would be nice a comment above it :)